### ISSUE TITLE
URLs are not properly generated for a set named "0" in function generateURLs from Utils.pm

### ISSUE THREAD

**Issue by somiaj** [openwebwork/webwork2#1323]
Hello,

I have installed a new server using the develop branch and have noticed the following when I was testing it out.

Emails sent using the 'Email Instructor' button from a problem in a homework set do not fully keep track of which problem the email was sent from. The link `$emailalbleURL` sent with the email only links to the user who sent the problem, and no longer links to the problem from which the email was sent from. In addition clicking `cancel email` or `return to your work` returns to the homework list and not the problem from which the email was sent from. The links I see are just `https://server/webwork2/Course/?effectiveUser=practice2` and no longer contain info about the problem the user was working on.

The email details contain the info about which set and problem the email was sent from, but this information is not being recorded in the links like I am use to on my other server.

**Comment by somiaj** [comment]
Okay, I think I've tracked down the actual issue, and for the most part it is working. When I was testing this, I decided to use the default `set0.def` to send an email from. It appears that if the set name is `0` that the links don't get created properly. But if I used a set with an actual name, here I used `setMAAtutorial`, the links all generate as expected. So there is some issue in generating the links of the set name is `0`.

I also tested this using a set named `1`, and that worked fine as well. So it seems this issue only affects any set that has the name `0`.

I have updated the issue title.
